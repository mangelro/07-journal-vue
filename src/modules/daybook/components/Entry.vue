<template>
     <div class="row no-wrap top-align divider entry-container">
            <div class="col min">
              <i class="light-green-text">check_circle</i>
            </div>
            <div class="col">
              <span>{{day}}</span>&nbsp;
              <span>{{month}}</span>&nbsp;
              <span>{{yearDate}}</span>
              <div>Title</div>
              <label>{{shortText}}</label>
            </div>
            <div class="col min">
              <nav>
                <button class="none" @click="$router.push({name:'entry',params:{id:entry.id}})">Ver</button>
              </nav>
            </div>
          </div>
</template>

<script>

import getDayMonthYear from '../helpers/getDayMonthYear'
export default {
  props:{
    entry:{
      type:Object,
      required:true,
    }
  },

  computed:{
    shortText(){
      return this.entry.text.length>130? this.entry.text.substring(0,130)+'...':this.entry.text
    },
    day(){
      const {day} = getDayMonthYear(this.entry.date)
      return day
    },
    month(){
      const {month} = getDayMonthYear(this.entry.date)
      return month
    },
    yearDate(){
      const {year} = getDayMonthYear(this.entry.date)
      return year
    }
  }

}
</script>

<style  lang="scss" scoped>

    .entry-container{
        transition:0.2s all ease-in
    }
    .entry-container:hover{
        background-color: lighten($color: grey, $amount: 45);
        transition:0.2s all ease-in
    }
</style>