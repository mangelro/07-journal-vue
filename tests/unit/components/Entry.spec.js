import {shallowMount} from '@vue/test-utils'

import Entry from '@/modules/daybook/components/Entry'


describe('Pruebas del Entry component',()=>{


	//mockRouter
	//const wrapper (entry,propt...global...)

	test('Debe hacer match con el snapshot', () => {
	
		const wrapper = shallowMount(Entry,{
			props:{
				entry:{
					id:"-Mwm0kple69InwR9BYO4",
					date : 1645811782393,
					text : "Estoy aprendiendo Vue"
				}
			}
		})

		expect(wrapper.html()).toMatchSnapshot()
	})

	test('Debe redireccional al hacer click en el entry-container', async () => {

		const mockRouter={
			push:jest.fn()
		}
		
		const wrapper = shallowMount(Entry,{
			props:{
				entry:{
					id:"-Mwm0kple69InwR9BYO4",
					date : 1645811782393,
					text : "Estoy aprendiendo Vue"
				}
			},
			global:{
				mocks:{
					$router:mockRouter
				}
			}
		})

		wrapper.find('button').trigger('click')

		
        expect(mockRouter.push).toHaveBeenCalled()
        expect(mockRouter.push).toHaveBeenCalledWith({name: 'entry',params: { id: '-Mwm0kple69InwR9BYO4'}})
	})

	test('Pruebas en la propiedades computadas de la fecha', () => {
		const wrapper = shallowMount(Entry,{
			props:{
				entry:{
					id:"-Mwm0kple69InwR9BYO4",
					date : 1645811782393,
					text : "Estoy aprendiendo Vue"
				}
			}
		})


		expect(wrapper.vm.day).toBe('25')
		expect(wrapper.vm.month).toBe('Febrero')
		expect(wrapper.vm.yearDate).toBe('2022, Viernes')

		//wrapper.vm. => ver las propiedades computadas
		//day = corresponde con al entrada
		//month = ....
		//yearDate:....
	})



})
