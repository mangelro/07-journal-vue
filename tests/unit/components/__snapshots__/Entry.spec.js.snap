// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del Entry component Debe hacer match con el snapshot 1`] = `
<div class="row no-wrap top-align divider entry-container">
  <div class="col min"><i class="light-green-text">check_circle</i></div>
  <div class="col"><span>25</span>&nbsp; <span>Febrero</span>&nbsp; <span>2022, Viernes</span>
    <div>Title</div><label>Estoy aprendiendo Vue</label>
  </div>
  <div class="col min">
    <nav><button class="none"> Ver </button></nav>
  </div>
</div>
`;
